tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1563,3 +1563,121 @@ TEST_F(EnergyPlusFixture, ShadowCalculation_CSV_extra_parenthesis)
+        "  Through: 4/30,",
+        "  For: WeekDays CustomDay1 CustomDay2,",
+        "  Until: 24:00,0.21,",
+        "  For: Weekends Holiday,",
+        "  Until: 24:00,0.65,",
+        "  For: SummerDesignDay,",
+        "  Until: 24:00,0.76,",
+        "  Until: 24:00,0.79,",
+        "  For: WinterDesignDay,",
+        "  Until: 24:00,0.16,",
+        "  Until: 24:00,0.18,",
+        "  Through: 12/31,",
+        "  For: WeekDays CustomDay1 CustomDay2,",
+        "  Until: 24:00,0.73,",
+        "  For: Weekends Holiday,",
+        "  Until: 24:00,0.27,",
+        "  For: SummerDesignDay,",
+        "  Until: 8:00,0.77,",
+        "  Until: 24:00,0.85,",
+        "  For: WinterDesignDay,",
+        "  Until: 8:00,0.19,",
+        "  Until: 24:00,0.25;",
+        " ",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 4;
+    state->dataGlobal->MinutesPerTimeStep = 15;
+    state->dataGlobal->TimeStepZone = 0.25;
+
+    Real64 schMin;
+    Real64 schMax;
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, -1, DayTypeGroup::Weekday);
+    EXPECT_EQ(1.0, schMin);
+    EXPECT_EQ(1.0, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, 0, DayTypeGroup::Weekday);
+    EXPECT_EQ(0.0, schMin);
+    EXPECT_EQ(0.0, schMax);
+
+    int index = GetScheduleIndex(*state, "HIGHLOW01");
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::Weekday);
+    EXPECT_EQ(0.40, schMin);
+    EXPECT_EQ(0.87, schMax);
+
+    // repeat test to see if using cached values
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::Weekday);
+    EXPECT_EQ(0.40, schMin);
+    EXPECT_EQ(0.87, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::WeekEndHoliday);
+    EXPECT_EQ(0.30, schMin);
+    EXPECT_EQ(0.83, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::SummerDesignDay);
+    EXPECT_EQ(0.85, schMin);
+    EXPECT_EQ(0.95, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::WinterDesignDay);
+    EXPECT_EQ(0.15, schMin);
+    EXPECT_EQ(0.17, schMax);
+
+    index = GetScheduleIndex(*state, "HIGHLOW02");
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::Weekday);
+    EXPECT_EQ(0.21, schMin);
+    EXPECT_EQ(0.73, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::WeekEndHoliday);
+    EXPECT_EQ(0.27, schMin);
+    EXPECT_EQ(0.65, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::SummerDesignDay);
+    EXPECT_EQ(0.76, schMin);
+    EXPECT_EQ(0.85, schMax);
+
+    std::tie(schMin, schMax) = getScheduleMinMaxByDayType(*state, index, DayTypeGroup::WinterDesignDay);
+    EXPECT_EQ(0.16, schMin);
+    EXPECT_EQ(0.25, schMax);
```

Comment:
It took me a minute to see the 2 WinterDesignDay in the HIGHLOW02 schedule. Same for other schedule types. Good unit test.